src/game.cpp

```diff
@@ -2180,16 +2188,27 @@ bool Game::connectToServer(const std::string &playername,
 				break;
 			}
 
-			wait_time += dtime;
-			// Only time out if we aren't waiting for the server we started
-			if (!address->empty() && wait_time > 10) {
-				*error_message = "Connection timed out.";
-				errorstream << *error_message << std::endl;
-				break;
-			}
+			if (client->m_registration_confirmation_state == true) {
+				if (registration_confirmation_shown == false) {
```

Comment:
please swap the bodies of this if statement, invert the condition, and remove comparisons against booleans.

ie: 

```cpp
if (registration_confirmation_shown) {
    // ---
} else {
   // ---
}
```